cluster.go

```diff
@@ -1021,6 +1040,29 @@ func (c *Cluster) StateSync(ctx context.Context) error {
 	return nil
 }
 
+// getClosestTrustedPeer gets the highest trusted peer that is smaller than cid.
+func getClosestTrustedPeer(ctx context.Context, cidStr string, peers []peer.ID, c Consensus) peer.ID {
+	var closest peer.ID
+	for _, p := range peers {
+		if !c.IsTrustedPeer(ctx, p) {
+			continue
+		}
+
+		if closest == "" {
+			closest = p
+			continue
+		}
+
+		if cidStr > peer.IDB58Encode(p) || closest > p {
+			continue
+		}
+
+		closest = p
+	}
+
+	return closest
+}
+
```

Comment:
Rework all this into something that makes sense. You are comparing something like a peer ID B58 encoding (which maybe looks like `Qm...` with the `String()` of a cid which may be looks like `ba...` or whatever. This comparison does not make any sense and you are likely to always result in the same peer ID. I have no clue what you are doing because there is an additional peer.ID comparison involved and that's happening on a stringified byte-slice.

The right way to do it is exemplified by the dht code: you have really useful functions here: https://github.com/libp2p/go-libp2p-kbucket/blob/master/util.go#L55 (usef them)

Remember to ConvertKey() before-hand on the CID so the actual two things that you are calculating the distance on are barely comparable.